pkg/driver/trace_driver.go

```diff
@@ -203,43 +224,57 @@ func (d *Driver) invokeFunction(metadata *InvocationMetadata) {
 	defer metadata.AnnounceDoneWG.Done()
 
 	var success bool
-
+	node := metadata.Node
 	var record *mc.ExecutionRecord
-	switch d.Configuration.LoaderConfiguration.Platform {
-	case "Knative":
-		success, record = InvokeGRPC(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			d.Configuration.LoaderConfiguration,
-		)
-	case "OpenWhisk":
-		success, record = InvokeOpenWhisk(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			metadata.AnnounceDoneExe,
-			metadata.ReadOpenWhiskMetadata,
-		)
-	case "AWSLambda":
-		success, record = InvokeAWSLambda(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			metadata.AnnounceDoneExe,
-		)
-	case "Dirigent":
-		success, record = InvokeDirigent(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			d.Configuration.LoaderConfiguration,
-		)
-	default:
-		log.Fatal("Unsupported platform.")
+	var runtimeSpecifications *common.RuntimeSpecification
+	for {
+		if d.Configuration.LoaderConfiguration.DAGMode {
+			runtimeSpecifications = &node.Function.Specification.RuntimeSpecification[metadata.MinuteIndex][metadata.InvocationIndex]
+		} else {
+			runtimeSpecifications = metadata.RuntimeSpecifications
```

Comment:
Do we need to pass the RuntimeSpecification if we can get them in a general way from the function specification inside the current node? I think we shouldn't have a condition on DAGMode in invocation, we consider this method always in this mode since we are passing the linked list of functions to it.

Reply:
In normal uses, we can get them. However, for the test function ```testInvokeFunctionFromDriver``` in 
```pkg/driver/trace_driver_test.go```, they do not initialize the ```Function.Specification``` field, which contains the RuntimeSpecification. Should I do so?